File: Trabalho_3_C#/Trabalho_3_CSharp/Uniao.cs
using System;

namespace Trabalho_3_CSharp {
    public class Uniao<T> : Conjunto<T> {

        private readonly Conjunto<T> conjunto1;
        private readonly Conjunto<T> conjunto2;

        public Uniao(Conjunto<T> conjunto1, Conjunto<T> conjunto2) {
            this.conjunto1 = conjunto1;
            this.conjunto2 = conjunto2;
        }

        public override Boolean contemElemento(T elemento) {
            return conjunto1.contemElemento(elemento) || conjunto2.contemElemento(elemento);
        }

    }
}

File: Trabalho_3/src/com/company/Uniao.java
/**
 * @file  Uniao.java
 * @brief  Arquivo dedicado à classe Uniao.
 * @author Gustavo Rangel Bandeira Cauduro
 * @date   2022-01-15
 */

package com.company;

public class Uniao<T> extends Conjunto<T> {

    private final Conjunto<T> conjunto1;
    private final Conjunto<T> conjunto2;

    public Uniao(Conjunto<T> conjunto1, Conjunto<T> conjunto2) {
        this.conjunto1 = conjunto1;
        this.conjunto2 = conjunto2;
    }

    @Override
    public Boolean contemElemento(T elemento) {
        return conjunto1.contemElemento(elemento) || conjunto2.contemElemento(elemento);
    }

}
